functions/git-info

```diff
@@ -107,15 +108,35 @@ if ! zstyle -t ':zim:git-info' verbose; then
   if [[ -n ${indexed_format} || (${dirty} -eq 0 && (-n ${dirty_format} || -n ${clean_format})) ]]; then
     cmds+=("print -n 'i:'; command git diff-index --no-ext-diff --quiet --cached --ignore-submodules=${ignore_submodules} HEAD &>/dev/null; print \${?}")
   fi
+
+  local added_format
+  zstyle -s ':zim:git-info:added' format 'added_format'
+  
+  local deleted_format
+  zstyle -s ':zim:git-info:added' format 'deleted_format'
+
+  local modified_format
+  zstyle -s ':zim:git-info:added' format 'modified_format'
+
+  local renamed_format
+  zstyle -s ':zim:git-info:added' format 'renamed_format'
 else
-  local -i indexed unindexed untracked
+  local -i indexed unindexed untracked added deleted modified renamed
   # Get current status.
   command git status --porcelain --ignore-submodules=${ignore_submodules} 2>/dev/null | while IFS= read -r line; do
     if [[ ${line:0:2} == '??' ]]; then
       (( untracked++ ))
     else
       if [[ ${line:0:1} != ' ' ]] (( indexed++ ))
       if [[ ${line:1:1} != ' ' ]] (( unindexed++ ))
+
+      if [[ ${line:0:1} == 'A' ]] ((added++))
+      if [[ ${line:0:1} == 'M' ]] ((modified++))
+      if [[ ${line:1:1} == 'M' ]] ((modified++))
+      if [[ ${line:0:1} == 'D' ]] ((deleted++))
+      if [[ ${line:1:1} == 'D' ]] ((deleted++))
+      if [[ ${line:0:1} == 'R' ]] ((renamed++))
```

Comment:
Note that this might be tricker that what's implemented here. Sorin came up with this at the time:

https://github.com/sorin-ionescu/prezto/blob/9195b66161b196238cbd52a8a4abd027bdaf5f73/modules/git/functions/git-info#L367-L375